cmd/juju/caas/add.go

```diff
@@ -223,13 +231,30 @@ func (c *AddCAASCommand) Init(args []string) (err error) {
 	if c.contextName != "" && c.clusterName != "" {
 		return errors.New("only specify one of cluster-name or context-name, not both")
 	}
-	if c.hostCloudRegion != "" || c.cloud != "" {
-		c.hostCloudRegion, err = c.tryEnsureCloudTypeForHostRegion(c.cloud, c.hostCloudRegion)
-		if err != nil {
-			return errors.Trace(err)
+
+	end := 1
+	if argsCount > 1 {
+		c.cloudRegion = args[1]
+		end = 2
+		goto next
+	}
+
+	// TODO (anastsiamac 2019-07-24) This is unnecessary when --region and --cloud is removed.
+	if c.hostCloudRegion != "" {
+		if c.cloud != "" {
+			if strings.Contains(c.cloud, "/") || strings.Contains(c.hostCloudRegion, "/") {
+				// <cloud>/<region> definitions may conflict.
+				return errors.New("provide either --region or --cloud, not both")
+			}
+			c.cloudRegion = jujucloud.BuildHostCloudRegion(c.cloud, c.hostCloudRegion)
+			goto next
 		}
+		c.cloudRegion = c.hostCloudRegion
+	} else {
+		c.cloudRegion = c.cloud
```

Comment:
It really is hostCloudRegion rather than cloudRegion as the intent is to record where the host nodes are running.